### PR TITLE
move all allocators to use redis allocator

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

**Comment by MeirShpilraien** [comment]
@mnunberg this is an important fix especially now when lots of modules starts to use rediseach we do not want all of our modules to gets fragmentation issues.
That said, lets not merge until I will check with valgrind.

**Comment by MeirShpilraien** [comment]
@mnunberg Valgrind passed, I guess the memory sanitizers are doing their job :)

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@ OPTION(RS_BUILD_STATIC "Builds a static library" OFF)
 OPTION(USE_ASAN "Use AddressSanitizer (clang)" OFF)
 OPTION(USE_MSAN "Use MemorySanitizer (clang)" OFF)
 OPTION(USE_TSAN "Use ThreadSanitizer (clang)" OFF)
+OPTION(REDIS_ALLOCATOR "Use redis allocator" ON)
```
Comment (mnunberg) [inline CMakeLists.txt]: `USE_REDIS_ALLOCATOR`

**Comment by MeirShpilraien** [comment]
@mnunberg done

**Pull request merged**